ERCS/erc-7806.md

```diff
@@ -0,0 +1,451 @@
+| userIntent[42:44]               | `uint16`   | Length of instructions                                          |
+| userIntent[44:46]               | `uint16`   | Length of signatures                                            |
+| Next `headerLength` bytes       | `bytes`    | The actual header                                               |
+| Next `instructionsLength` bytes | `bytes`    | The actual instructions                                         |
+| Next `signatureLength` bytes    | `bytes`    | The actual signatures                                           |
+| Anything left                   | `bytes`    | Extra data, e.g. nested intent to be executed within the intent |
+
+The above schema is NOT REQUIRED for all implementations. Each `IAccount` and `IStandard` can fully determine their own way of handling the `bytes` data. In the above example, the fields can be used in this way
+* `header`: The `bytes header` can carry information about how to validate the intent or how to prevent
+  double-spending. For example, `header` can contain an `uint256 nonce` to check if the `nonce` is used already.
+* `instructions`: These `bytes instructions` can just be concatenated `(address,value,calldata)` or can be
+  standardized values, for example `(erc20TokenAddress,1000)` means the `instructions` can use up to 1000 of the
+  specified [ERC-20](./eip-20.md) token. It is NOT REQUIRED that all `instructions` MUST be provided by the EOA owner to allow dynamically carry out other operations during intent executions, but the `IStandard` design needs to carefully handle this case.
+* `signatures`: The `bytes signatures` field can support different signing methods. It is NOT REQUIRED that
+  all `signatures` MUST be provided by the EOA owner, some of them MAY be provided by solver, relayer or anyone else.
+
+### IStandard interface
+
+Each standard defines how to parse and validate a user intent, it implements the following `IStandard` interface:
+
+```solidity
+interface IStandard {
+    /**
+     * Validate user's intent
+     *
+     * @dev returning validation result, the type uses bytes4 for extensibility purpose
+     * @return result values representing validation outcomes
+     */
+    function validateUserIntent(bytes calldata intent) external view returns (bytes4 result);
+
+    /**
+     * Unpack user's intent, it is RECOMMENDED to validate intent while unpacking to save gas
+     *
+     * @dev returning unpacked result, the type uses bytes for extensibility purpose
+     * @return result unpacked result status
+     * @return operations unpacked operations that can be executed by the IAccount, NOT REQUIRED to match UserIntent.instructions
+     */
+    function unpackOperations(bytes calldata intent) external view returns (bytes4 memory result, bytes[] memory operations);
+}
+```
+
+Notice that `IStandard` fully controls the validation process, similar to `EntryPoint` in ERC-4337 and ERC-7521,
```

Comment:
> [...] be pre-audited [...]

We try to keep standards focused on the observable behaviour of the contracts. Auditing and testing and the like aren't necessary for a contract to be compliant with the standard (and aren't enforceable at all) so you should leave them out of the document.

Reply:
Updated 👍